wkcuber/api/layer.py

```diff
@@ -223,37 +368,33 @@ def set_bounding_box(
         """
         Updates the offset and size of the bounding box of this layer in the properties.
         """
-        self.dataset.properties._set_bounding_box_of_layer(self.name, offset, size)
-        bounding_box = BoundingBox(offset, size)
+        self._properties.bounding_box = BoundingBox(offset, size)
 
         for mag, mag_view in self.mags.items():
-            mag_view.size = cast(
+            mag_view._size = cast(
                 Tuple[int, int, int],
                 tuple(
-                    bounding_box.align_with_mag(mag, ceil=True).in_mag(mag).bottomright
+                    self._properties.bounding_box.align_with_mag(mag, ceil=True)
+                    .in_mag(mag)
+                    .bottomright
                 ),
             )
+        self.dataset._export_as_json()
 
     def set_bounding_box_offset(self, offset: Tuple[int, int, int]) -> None:
         """
         Updates the offset of the bounding box of this layer in the properties.
         """
-        size: Tuple[int, int, int] = self.dataset.properties.get_bounding_box_of_layer(
-            self.name
-        )[1]
-        self.set_bounding_box(offset, size)
+        self.set_bounding_box(offset, self._properties.bounding_box.size)
 
     def set_bounding_box_size(self, size: Tuple[int, int, int]) -> None:
         """
         Updates the size of the bounding box of this layer in the properties.
         """
-        offset: Tuple[
-            int, int, int
-        ] = self.dataset.properties.get_bounding_box_of_layer(self.name)[0]
-        self.set_bounding_box(offset, size)
+        self.set_bounding_box(self._properties.bounding_box.topleft, size)
 
     def get_bounding_box(self) -> BoundingBox:
-        return self.dataset.properties.data_layers[self.name].get_bounding_box()
+        return self._properties.bounding_box.copy()
 
     def rename(self, layer_name: str) -> None:
```

Comment:
Ah, this might actually just become the setter I mentioned above.

Reply:
Yes